reactor-core/src/main/java/reactor/util/context/ContextN.java

```diff
@@ -36,24 +35,35 @@
 			Object key3, Object value3, Object key4, Object value4,
 			Object key5, Object value5, Object key6, Object value6) {
 		super(6, 1f);
-		super.put(key1, value1);
-		super.put(key2, value2);
-		super.put(key3, value3);
-		super.put(key4, value4);
-		super.put(key5, value5);
-		super.put(key6, value6);
+		innerPut(key1, value1);
+		innerPut(key2, value2);
+		innerPut(key3, value3);
+		innerPut(key4, value4);
+		innerPut(key5, value5);
+		innerPut(key6, value6);
 	}
 
 	ContextN(Map<Object, Object> map, Object key, Object value) {
 		super(map.size() + 1, 1f);
-		super.putAll(map);
-		super.put(key, value);
+		innerPutAll(map);
+		innerPut(key, value);
 	}
 
 	ContextN(Map<Object, Object> sourceMap, Map<?, ?> other) {
 		super(sourceMap.size() + other.size(), 1f);
-		super.putAll(sourceMap);
-		super.putAll(other);
+		innerPutAll(sourceMap);
+		innerPutAll(other);
+	}
+
+	private void innerPut(Object key, Object value) {
+		super.put(Objects.requireNonNull(key, "null key passed to constructor"),
+				Objects.requireNonNull(value, "null value passed to constructor"));
+	}
+
+	private void innerPutAll(Map<?, ?> map) {
+		Objects.requireNonNull(map, "map in constructor")
+		       .forEach((k,v) -> super.put(Objects.requireNonNull(k, "map in constructor contains null key"),
```

Comment:
why not just `.forEach(this::innerPut)`? The stacktrace will point to the constructor anyways, and we can remove "passed to constructor" from `innerPut`

Reply:
yeah this won't provide more information that what the stacktrace does, you're right